agents-and-function-calling/bedrock-agents/features-examples/11-get-user-confirmation-before-action-execution/11.1-create-agent-with-function-definition-and-user-confirmation/11.1-create-agent-with-function-definition-and-user-confirmation.ipynb

```diff
@@ -0,0 +1,2222 @@
+     ]
+    }
+   ],
+   "source": [
+    "raw_response_with_roc_allowed = bedrock_agent_runtime_client.invoke_agent(\n",
+    "    agentId=agent_id,\n",
+    "    agentAliasId=agent_alias_id, \n",
+    "    sessionId=session_id,\n",
+    "    enableTrace=enable_trace, \n",
+    "    sessionState={\n",
+    "        'invocationId': event[\"returnControl\"][\"invocationId\"],\n",
+    "        'returnControlInvocationResults': [{\n",
+    "                'functionResult': {\n",
+    "                    'actionGroup': event[\"returnControl\"][\"invocationInputs\"][0][\"functionInvocationInput\"][\"actionGroup\"],\n",
+    "                    'function': event[\"returnControl\"][\"invocationInputs\"][0][\"functionInvocationInput\"][\"function\"],\n",
+    "                    'confirmationState': 'CONFIRM',\n",
+    "                    'responseBody': {\n",
+    "                        \"TEXT\": {\n",
+    "                            'body': str(response_reserve_vacation_time[\"response\"][\"functionResponse\"][\"responseBody\"][\"TEXT\"][\"body\"])\n",
+    "                        }\n",
+    "                    }\n",
+    "                }\n",
+    "        }]}\n",
+    ")\n",
+    "\n",
+    "print(raw_response_with_roc_allowed)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 207,
+   "id": "9ee18921",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'chunk': {'bytes': b'One day of vacation time on June 1, 2024 has been successfully reserved for employee with id 1.'}}\n",
+      "CPU times: user 1.9 ms, sys: 0 ns, total: 1.9 ms\n",
+      "Wall time: 2.29 ms\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "event_stream = raw_response_with_roc_allowed['completion']\n",
+    "for event_allowed in event_stream:\n",
+    "    print(event_allowed)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 212,
+   "id": "8a7a3c4c",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'returnControl': {'invocationId': '918c5b9b-9b02-4567-b382-0ffb5bd7c70e-uc', 'invocationInputs': [{'functionInvocationInput': {'actionGroup': 'VacationsActionGroup', 'actionInvocationType': 'USER_CONFIRMATION', 'function': 'reserve_vacation_time', 'parameters': [{'name': 'end_date', 'type': 'string', 'value': '2024-06-01'}, {'name': 'employee_id', 'type': 'integer', 'value': '1'}, {'name': 'start_date', 'type': 'string', 'value': '2024-06-01'}]}}]}}\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(event)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "95aaab55",
+   "metadata": {},
+   "source": [
+    "## Invoking an agent with function results and execution is not allowed"
```

Comment:
why are we passing the function results? that makes no sense! we should let the lambda execute the function and just pass the confirmation